node/header_exchange.go

```diff
@@ -175,9 +178,10 @@ func (hExService *HeaderExchangeService) Start() error {
 			}
 		}
 	}
-	go hExService.tryInitHeaderStoreAndStartSyncer(hExService.ctx, trustedHeader)
-
-	return nil
+	if trustedHeader == nil {
+		return nil
+	}
+	return hExService.initHeaderStoreAndStartSyncer(hExService.ctx, trustedHeader)
```

Comment:
apply the same comments as in block exchange service. move `initHeaderStoreAndStartSyncer ` inside the outer-if.